federationapi/routing/threepid.go

```diff
@@ -89,7 +89,7 @@ func CreateInvitesFrom3PIDInvites(
 	}
 
 	// Send all the events
-	if err := api.SendEvents(req.Context(), rsAPI, api.KindNew, evs, cfg.Matrix.ServerName, nil, false); err != nil {
+	if err := api.SendEvents(req.Context(), rsAPI, api.KindNew, evs, "TODO", cfg.Matrix.ServerName, nil, false); err != nil {
```

Comment:
Yes, because for some reason this endpoint is `MakeExternalAPI` instead of `MakeFedAPI` and therefore there's no `gomatrixserverlib.FederationRequest` to tell us what the real origin of the request was, and I haven't looked into why yet.